backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt

```diff
@@ -640,4 +648,48 @@ class ShorkInterpreterControllerV1IT : AbstractControllerTest() {
             assertTrue(response.errors.isNotEmpty())
         }
     }
+
+    @Nested
+    inner class GetLobbySettings {
+        @Test
+        fun `test get lobby settings for an existing lobby`() = runTest {
+            val clientLobby =
+                client.post("/api/v1/lobby") {
+                    contentType(ContentType.Application.Json)
+                    setBody("{\"playerName\":\"playerA\"}")
+                }
+            val responseId =
+                Json.parseToJsonElement(clientLobby.bodyAsText())
+                    .jsonObject["lobbyId"]!!
+                    .jsonPrimitive
+                    .content
+            val result = client.get("/api/v1/lobby/$responseId/settings")
+            val parsedSettings = parseSettings(result)
+            val defaultSettings = Settings()
+
+            assertEquals(HttpStatusCode.OK, result.status)
+            assertTrue(
+                parsedSettings ==
+                    InterpreterSettings(
+                        coreSize = defaultSettings.coreSize,
+                        instructionLimit = defaultSettings.instructionLimit,
+                        initialInstruction = defaultSettings.initialInstruction,
+                        maximumTicks = defaultSettings.maximumTicks,
+                        maximumProcessesPerPlayer = defaultSettings.maximumProcessesPerPlayer,
+                        readDistance = defaultSettings.readDistance,
+                        writeDistance = defaultSettings.writeDistance,
+                        minimumSeparation = defaultSettings.minimumSeparation,
+                        separation = defaultSettings.separation,
+                        randomSeparation = defaultSettings.randomSeparation,
+                    )
+            )
+        }
+
+        @Test
+        fun `test get lobby settings for a non-existing lobby`() = runTest {
+            val result = client.get("/api/v1/lobby/0/settings")
+            assertEquals(HttpStatusCode.InternalServerError, result.status)
+            assertTrue(result.bodyAsText().contains("No lobby with that id"))
```

Comment:
It would be nice to not assert the message content, asserting beforehand that the Status is `404 NOT_FOUND` should be enough.